v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -82,4 +82,339 @@ typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_056: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint16_t).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_057: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint16_t).copy` shall copy the bytes of the `uint16_t` value from the address pointed by `src_value` to `dst_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_058: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint16_t).copy` shall succeed and return 0. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint16_t).free
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* value);
+```
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_059: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint16_t).free` shall return. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint16_t).get_type
+
+```c
+typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
+```
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_060: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint16_t).get_type` shall returns the property type `LOG_CONTEXT_PROPERTY_TYPE_uint16_t`. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(void* property_value, char* buffer, size_t buffer_length);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).to_string` produces the string representation of an `int8_t`.
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_061: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).to_string` shall fail and return -1. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_062: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).to_string` shall call `snprintf` with `buffer`, `buffer_length` and format string `PRId8` and pass in the values list the `int8_t` value pointed to be `property_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_063: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).to_string` shall succeed and return the result of `snprintf`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_064: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).to_string` shall fail and return -1. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).copy
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* dst_value, void* src_value);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).copy` copies the `int8_t` value from the address pointed by `src_value` to `dst_value`.
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_065: [** If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_066: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_067: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).copy` shall copy the bytes of the `int8_t` value from the address pointed by `src_value` to `dst_value`. **]**
```

Comment:
>bytes [](http://example.com/codeflow?start=112&length=5)

there's just 1 byte here :) a clear use for singular!

Reply:
Changed